main/src/main/scala/sbt/internal/SysProp.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+package internal
+
+import java.util.Locale
+import scala.util.control.NonFatal
+import sbt.internal.util.ConsoleAppender
+
+// See also BuildPaths.scala
+// See also LineReader.scala
+object SysProp {
+  def booleanOpt(name: String): Option[Boolean] =
+    sys.props.get(name).flatMap { x =>
+      x.toLowerCase(Locale.ENGLISH) match {
+        case "1" | "always" | "true" => Some(true)
+        case "0" | "never" | "false" => Some(false)
+        case "auto"                  => None
+        case _                       => None
+      }
+    }
+
+  def getOrFalse(name: String): Boolean = booleanOpt(name).getOrElse(false)
+  def getOrTrue(name: String): Boolean = booleanOpt(name).getOrElse(true)
+
+  def long(name: String, default: Long): Long =
+    sys.props.get(name) match {
+      case Some(str) =>
+        try {
+          str.toLong
+        } catch {
+          case NonFatal(_) => default
+        }
+      case _ => default
+    }
+
+  def int(name: String, default: Int): Int =
+    sys.props.get(name) match {
+      case Some(str) =>
+        try {
+          str.toInt
+        } catch {
+          case NonFatal(_) => default
+        }
+      case _ => default
+    }
+
+  // System property style:
+  //   1. use sbt. prefix
+  //   2. prefer short nouns
+  //   3. use dot for namespacing, and avoid making dot-separated English phrase
+  //   4. make active/enable properties, instead of "sbt.disable."
+  //
+  // Good: sbt.offline
+  //
+  // Bad:
+  // sbt.disable.interface.classloader.cache
+  // sbt.task.timings.on.shutdown
+  // sbt.skip.version.write -> sbt.genbuildprops=false
+
+  def offline: Boolean = getOrFalse("sbt.offline")
+  def traces: Boolean = getOrFalse("sbt.traces")
+  def client: Boolean = getOrFalse("sbt.client")
+  def ci: Boolean = getOrFalse("sbt.ci")
+
+  def watchMode: String =
+    sys.props.get("sbt.watch.mode").getOrElse("auto")
+
+  def residentLimit: Int = int("sbt.resident.limit", 0)
+
+  /**
+   * Indicates whether formatting has been disabled in environment variables.
+   * 1. -Dsbt.log.noformat=true means no formatting.
+   * 2. -Dsbt.color=always/auto/never/true/false
+   * 3. -Dsbt.colour=always/auto/never/true/false
+   * 4. -Dsbt.log.format=always/auto/never/true/false
+   */
+  lazy val color: Boolean = ConsoleAppender.formatEnabledInEnv
+
+  def supershell: Boolean = color && getOrTrue("sbt.supershell")
+
+  def supersheelSleep: Long = long("sbt.supershell.sleep", 100L)
```

Comment:
I think there's one `e` too many. Should be: supershellSleep

Reply:
Should've been supersheepSleep. 🐑🐑

Reply:
Looks like it's fixed in https://github.com/sbt/sbt/pull/4786/commits/a6bc7b1c7664c83a788878a6a76dd3b67e995d53